app/controllers/spree/admin/products_controller.rb

```diff
@@ -76,7 +76,7 @@ def bulk_update
 
         product_set.collection.each { |p| authorize! :update, p }
 
-        if product_set.save
+        if product_set.save && product_set.errors.empty?
```

Comment:
I think it would be better if `#save` returned `false` in this case. We could add a line to the end of this method:
https://github.com/openfoodfoundation/openfoodnetwork/blob/6f228781d47552eabf2de7a1a5c020504a9d28ca/app/services/sets/product_set.rb#L96-L100

```rb
product.errors.any?
```

But this works, too. It's just not quite the Rails convention of how `save` works.

Reply:
`product_set.save` returns `true` event if it has errors (I added the line on `update_variants_attributes ` method)

Reply:
Anyway, I think it's ok with `empty?`, here's the commit: 6f228781d

Reply:
Moving to `Test Ready` then ;)

Reply:
Ah, yes, perfect!